lib/src/transformer/declaration_parsing.dart

```diff
@@ -280,6 +275,29 @@ class ParsedDeclarations {
       }
     }
 
+    // validate that the factory is initialized correctly
+    final factory = declarationMap[key_factory].length <= 1 ? singleOrNull(declarationMap[key_factory]) : null;
```

Comment:
FYI `factory` just shows up weirdly formatted in GitHub, and isn't officially discouraged for use a variable name since it's a [built-in identifier](https://www.dartlang.org/guides/language/language-tour#keywords).

Reply:
FYI `factory` just shows up weirdly formatted in GitHub, and isn't officially discouraged for use a variable name since it's a [built-in identifier](https://www.dartlang.org/guides/language/language-tour#keywords).

Reply:
Oh, TIL! Thanks Greg :)